From e87f3b59c9434799786a2adaca671e5674d0b348 Mon Sep 17 00:00:00 2001
From: "J. Ryan Stinnett" <jryans@gmail.com>
Date: Fri, 13 Jun 2025 15:37:37 +0100
Subject: [PATCH 3/5] Add test suite dependencies to Nix flake

---
 flake.nix | 1 +
 1 file changed, 1 insertion(+)

diff --git a/flake.nix b/flake.nix
index 34affe757d..617bdcb625 100644
--- a/flake.nix
+++ b/flake.nix
@@ -27,16 +27,17 @@
         { pkgs }:
         {
           default = pkgs.mkShell {
             packages = with pkgs; [
               curl
               expat
               gettext
               openssl
+              perl
               zlib
             ];
             hardeningDisable = [ "fortify" ];
           };
         }
       );
     };
 }
-- 
2.49.0

